### PR TITLE
Yaml cleanup

### PR DESCRIPTION
**Pull request by Itxaka**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: overlay/files/system/oem/00_rootfs.yaml [hunk 1]
```diff
@@ -17,14 +17,7 @@ stages:
...
       environment:
-        VOLUMES: "LABEL=COS_OEM:/oem LABEL=COS_PERSISTENT:/usr/local"
-        OVERLAY: "tmpfs:25%"
-    - if: '[ -f "/run/cos/recovery_mode" ]'
-      # omit the persistent partition on recovery mode
-      name: "Layout configuration"
-      environment_file: /run/cos/cos-layout.env
-      environment:
-        VOLUMES: "LABEL=COS_OEM:/oem"
+        VOLUMES: "LABEL=COS_PERSISTENT:/usr/local"
```
Comment (Itxaka) [inline overlay/files/system/oem/00_rootfs.yaml:20]: oem is passed on the cmdlin

File: overlay/files/system/oem/21_kcrypt.yaml [hunk 2]
```diff
@@ -1,10 +1,5 @@
 name: "Kcrypt"
 stages:
-  rootfs:
```
Comment (Itxaka) [inline overlay/files/system/oem/21_kcrypt.yaml:3]: kcrypt is run by immucore

File: overlay/files/system/oem/25_default_paths.yaml [hunk 3]
```diff
@@ -8,3 +8,5 @@ stages:
         - path: /var/lib/snapd
         - path: /snap
         - path: /var/snap
+    - name: "Default system dirs (ubuntu)"
+      if: "[ $(kairos-agent state get oem.found) == 'true' ]"
```
Comment (Itxaka) [inline overlay/files/system/oem/25_default_paths.yaml:12]: woops

File: overlay/files/system/oem/03-setupcon.yaml [hunk 4]
```diff
@@ -2,6 +2,7 @@ name: "Start installer on tty1"
 stages:
   initramfs.after:
     - name: "setupcon initramfs.after"
+      if: "[[ $(kairos-agent state get kairos.flavor) =~ ^ubuntu ]]"
```
Comment (mauromorales) [inline overlay/files/system/oem/03-setupcon.yaml:5]: nice!

Comment (mauromorales) [reply overlay/files/system/oem/03-setupcon.yaml:5]: should we put the flavor in the name of the stage? just for when reading logs?

**Review by mauromorales — DISMISSED** [review]
sweet!

File: Earthfile [hunk 5]
```diff
@@ -215,12 +215,8 @@ framework:
     # Copy flavor-specific overlay files
     IF [[ "$FLAVOR" =~ ^alpine* ]]
         COPY overlay/files-alpine/ /framework
-    ELSE IF [ "$FLAVOR" = "fedora" ] || [ "$FLAVOR" = "rockylinux" ]
-        COPY overlay/files-fedora/ /framework
```
Comment (jimmykarily) [inline Earthfile:219]: we don't need the fedora files anymore?

Comment (Itxaka) [reply Earthfile:219]: probably yes, trying to find out if selinux works now on fedora+ubuntu....seems to work on some ubuntus somehow?

File: overlay/files/system/oem/25_default_paths.yaml [hunk 6]
```diff
@@ -1,10 +0,0 @@
-name: "default-paths"
-stages:
-  fs.before:
-    - name: "Default system dirs"
-      directories:
-        - path: /var/lib/longhorn
```
Comment (Itxaka) [inline overlay/files/system/oem/25_default_paths.yaml:6]: paths already created by the layout config

File: overlay/files/system/oem/27_harderning.yaml [hunk 7]
```diff
@@ -1,13 +0,0 @@
-name: "Security configuration hardening"
-stages:
-  initramfs:
-  - name: "Ensure runtime permission"
```
Comment (Itxaka) [inline overlay/files/system/oem/27_harderning.yaml:4]: moved to the accounting config

File: overlay/files/system/oem/06_recovery.yaml [hunk 8]
```diff
@@ -1,30 +0,0 @@
...
-     files:
-     - path: /run/cos/recovery_mode
-       content: "1"
-       permissions: 0600
-       owner: 0
-       group: 0
-   boot:
-   - name: "Recovery"
-     if: '[ -f "/run/cos/recovery_mode" ]'
-     hostname: "cos-recovery"
```
Comment (Itxaka) [inline overlay/files/system/oem/06_recovery.yaml:23]: merged with the recovery config

File: overlay/files/system/oem/32_journal_persistency.yaml [hunk 9]
```diff
@@ -1,7 +0,0 @@
-name: "Journalctl persistence"
-stages:
-  initramfs:
-    - name: "Create journalctl /var/log/journal dir"
-      if: '[ -e "/sbin/systemctl" ] || [ -e "/bin/systemctl" ] || [ -e "/usr/sbin/systemctl" ] || [ -e "/usr/bin/systemctl" ]'
```
Comment (Itxaka) [inline overlay/files/system/oem/32_journal_persistency.yaml:5]: moved to rootfs config

**Review by mauromorales — APPROVED** [review]

**Pull request merged**